src/graph/executor/algo/AllPathsExecutor.cpp

```diff
@@ -428,28 +320,27 @@ AllPathsExecutor::doBuildPath(size_t step,
   }
 
   auto newPathsSize = newPathsPtr->size();
-  if (step > maxStep_ || newPathsSize == 0) {
-    return folly::makeFuture<std::vector<std::pair<NPath*, Value>>>(std::move(*currentStepResult));
+  if (newPathsSize == 0) {
+    return folly::makeFuture<std::vector<NPath*>>(std::vector<NPath*>());
   }
-  std::vector<folly::Future<std::vector<std::pair<NPath*, Value>>>> futures;
+  std::vector<folly::Future<std::vector<NPath*>>> futures;
   if (newPathsSize < FLAGS_path_batch_size) {
-    futures.emplace_back(folly::via(runner(), [this, step, newPathsSize, newPathsPtr]() {
-      return doBuildPath(step + 1, 0, newPathsSize, newPathsPtr);
+    futures.emplace_back(folly::via(runner(), [this, step, newPathsSize, newPathsPtr, reverse]() {
+      return doBuildPath(step + 1, 0, newPathsSize, newPathsPtr, reverse);
     }));
   } else {
     for (size_t _start = 0; _start < newPathsSize; _start += FLAGS_path_batch_size) {
       auto tmp = _start + FLAGS_path_batch_size;
       auto _end = tmp > newPathsSize ? newPathsSize : tmp;
-      futures.emplace_back(folly::via(runner(), [this, step, _start, _end, newPathsPtr]() {
-        return doBuildPath(step + 1, _start, _end, newPathsPtr);
+      futures.emplace_back(folly::via(runner(), [this, step, _start, _end, newPathsPtr, reverse]() {
+        return doBuildPath(step + 1, _start, _end, newPathsPtr, reverse);
```

Comment:
not familiar with the logic here, should  _start  also be updated?

Reply:
each iteration _start in the for statement will be updated